### ISSUE TITLE
[jacobsen on 2009-09-01 14:35:45] : ClusterTrigger causes problems w/ flasher runs

### ISSUE THREAD

**Issue by dglo** [dglo/pdaq#1165]
From my email summary to Chris W. and daq-dev:

DAQ flasher runs

We were able to get some improvement by reducing the flasher rate from 10 Hz to 3 Hz, and dramatic improvement by removing the ClusterTrigger (aka "String Trigger").

We extended the initial 5 minute tests to 15 minutes:

- Run 114419: scan_test_3 (10Hz), ran almost to completion (15 minutes).  Died because of Cluster Trigger bug about 20 seconds before the end of the run.

- Run 114421: scan_test_4 (10Hz) ran for about 10 minutes. Died because of Cluster Trigger bug.

- Run 114422 and 114424: scan_test_5 (10Hz) and scan_test_6 (10Hz) died almost immediately due to the Cluster Trigger bug.

- Run 114429 - no Cluster trigger: scan_test_6_lorate (3Hz) ran fine to completion.

- Run 114431 - no Cluster trigger: started scan_test_6_lorate (3Hz), stopped that and started scan_test_6 (10Hz).  Ran to completion.

In short, we determined that removing the Cluster Trigger (aka "string trigger") allowed the most difficult test to complete successfully.  It was previously seen that Cluster Trigger sometimes caused runs to fail directly (Java stacktrace).  The success of the last few runs may simply have been because, without the Cluster Trigger, the overall trigger rate was reduced by about 20%.  Or it may be that issues in the Cluster Trigger caused the downstream problems Dave saw in previous runs.

**Comment by dglo** [comment]
[kaeld on 2009-09-14 17:43:14]
checkin 4589 fixes a problem with zero-length hit lists getting passed into formTrigger( ) which seems to have been causing this mischief

checkin 4590 adds a check in formTrigger() so this cannot happen again\n\n
